### DISCUSSION TITLE
Difference between vtkView and vtkLocalView/vtkRemoteView

### DISCUSSION THREAD

**Discussion by pazars** [Kitware/trame#100] [Q&A] [answered]
While most examples as well as the tutorial discuss the use of vtkLocalView and vtkRemoteView, there are examples like `06_vtk/Applications/SurfacePicking` that use a different approach to setup the vtk pipeline with vtkView. 

Is the vtkView intended for some specific use case? What are the advantages and disadvantages of using each approach?

**Comment by jourdain** [comment, marked as answer]
As a quick answer you can think of:

- VtkLocalView and VtkRemoteView rely on VTK/C++ and some vtk.js for the client side.
- VtkView is just plain vtk.js which means you could create an app that does not involved VTK/C++ on the server side by driving the code more on the client side point of view.

HTH
